src/server/plugins/engine/components/ComponentCollection.ts

```diff
@@ -152,7 +152,16 @@ export class ComponentCollection {
   }
 
   get keys() {
-    return this.fields.map(({ name }) => name)
+    return this.fields.flatMap((field) => {
+      const { name, collection } = field
+
+      if (collection) {
+        const { fields } = collection
+        return [name, ...fields.map(({ name }) => name)]
+      }
+
+      return [name]
+    })
```

Comment:
We're now able to access `page.keys` so the top-level `page.collection` gathers nested names too